ecs/command/event-generator/event_generator.py

```diff
@@ -24,34 +26,48 @@
 urllib3.disable_warnings(urllib3.exceptions.InsecureRequestWarning)
 
 
-def generate_random_command(include_all_fields=False):
-    document = {
-        "source": random.choice(["Users/Services", "Engine", "Content manager"]),
-        "user": f"user{random.randint(1, 100)}",
-        "target": {
-            "id": f"target{random.randint(1, 10)}",
-            "type": random.choice(["agent", "group", "server"])
-        },
-        "action": {
-            "name": random.choice(["restart", "update", "change_group", "apply_policy"]),
-            "args": [f"/path/to/executable/arg{random.randint(1, 10)}"],
-            "version": f"v{random.randint(1, 5)}"
-        },
-        "timeout": random.randint(10, 100)
-    }
+def generate_random_date(initial_date=None, days_range=30):
+    if initial_date is None:
+        initial_date = datetime.datetime.now()
+    random_days = random.randint(0, days_range)
+    new_timestamp = initial_date + datetime.timedelta(days=random_days)
+    return int(new_timestamp.timestamp() * 1000)  # Convert to milliseconds and return as int
+
 
+def generate_random_command(include_all_fields=False):
+    document = {}
     if include_all_fields:
-        document["agent"]["groups"] = [f"group{random.randint(1, 5)}"],
-        document["command"]["status"] = random.choice(
-            ["pending", "sent", "success", "failure"])
-        document["command"]["result"] = {
+        document["@timestamp"] = generate_random_date()
+        document["delivery_timestamp"] = generate_random_date()
+        document["agent"] = {"groups": [f"group{random.randint(1, 5)}"]}
+        document["command"] = {
+          "status": random.choice(["pending", "sent", "success", "failure"]),
+          "result": {
             "code": random.randint(0, 255),
             "message": f"Result message {random.randint(1, 1000)}",
             "data": f"Result data {random.randint(1, 100)}"
+          },
+          "request_id": str(uuid.uuid4()),
+          "order_id": str(uuid.uuid4())
         }
         # Generate UUIDs for request_id and order_id
         document["command"]["request_id"] = str(uuid.uuid4())
         document["command"]["order_id"] = str(uuid.uuid4())
+    else:
+        document = {
+          "source": random.choice(["Users/Services", "Engine", "Content manager"]),
+          "user": f"user{random.randint(1, 100)}",
+          "target": {
+            "id": f"target{random.randint(1, 10)}",
+            "type": random.choice(["agent", "group", "server"])
+          },
+          "action": {
+            "name": random.choice(["restart", "update", "change_group", "apply_policy"]),
+            "args": [f"/path/to/executable/arg{random.randint(1, 10)}"],
+            "version": f"v{random.randint(1, 5)}"
+          },
+          "timeout": random.randint(10, 100)
+        }
```

Comment:
Timeout field is missing if `include_all_fields` is set. I think the previous approach where the data was expanded instead of using a different logic depending on this value was better, as it is easier to see which fields are inputs of the API and which are generated by the CM plugin and added to the document. 

https://github.com/wazuh/wazuh-indexer/blob/abba78d5d858b80baac59a33a1a219969e92b56a/ecs/command/event-generator/event_generator.py#L27-L56

Reply:
`@timestamp` and `delivery_timestamp` should be added under `if include_all_fields:`, as other fields generated by the CM plugin.

Reply:
Function `generate_random_command` updated on commit https://github.com/wazuh/wazuh-indexer/pull/568/commits/3a0ffb057d944ce35b7ea4c305655a46b49f3eef.
Now the _non-all-fields_ request generates the command body, this command is included into the _document_ for the _all-fields_ case in the `command` field.

```python
def generate_random_command(include_all_fields=False):
    command = {
        "source": random.choice(["Users/Services", "Engine", "Content manager"]),
        "user": f"user{random.randint(1, 100)}",
        "target": {
            "id": f"target{random.randint(1, 10)}",
            "type": random.choice(["agent", "group", "server"])
        },
        "action": {
            "name": random.choice(["restart", "update","change_group", "apply_policy"]),
            "args": [f"/path/to/executable/arg{random.randint(1, 10)}"],
            "version": f"v{random.randint(1, 5)}"
        },
        "timeout": random.randint(10, 100)
    }
    if include_all_fields:
        document = {
            "@timestamp": generate_random_date(),
            "delivery_timestamp": generate_random_date(),
            "agent": {"groups": [f"group{random.randint(1, 5)}"]},
            "command": {
                **command,
                "status": random.choice(["pending", "sent", "success", "failure"]),
                "result": {
                    "code": random.randint(0, 255),
                    "message": f"Result message {random.randint(1, 1000)}",
                    "data": f"Result data {random.randint(1, 100)}"
                },
                "request_id": str(uuid.uuid4()),
                "order_id": str(uuid.uuid4())
            }
        }
        return document

    return command
```